### PR TITLE
Namgold/fix-token-select

### PR DESCRIPTION
**Pull request by namgold**
Signed-off-by: Nam Nguyen <nam.nguyen@kyber.network>

### CODE REVIEW

File: src/utils/tokenInfo.ts [hunk 1]
```diff
@@ -41,3 +42,11 @@ export const checkPairInWhiteList = (chainId: ChainId | undefined, symbol1: stri
     : ''
   return { isInWhiteList, data: data || {}, canonicalUrl }
 }
+
+export const getFormattedAddress = (address?: string, fallback?: string): string => {
```
Comment (viet-nv) [inline src/utils/tokenInfo.ts:46]: why do you need fallback here?

Comment (namgold) [reply src/utils/tokenInfo.ts:46]: additional option for anyone use it wanna fallback by custom string. Im not using it anw

File: src/state/lists/hooks.ts [hunk 2]
```diff
@@ -55,11 +56,11 @@ function listToTokenMap(list: TokenList): TokenAddressMap {
 
   const map = list.tokens.reduce<TokenAddressMapWriteable>(
     (tokenMap, tokenInfo) => {
-      if (tokenMap[tokenInfo.chainId][tokenInfo.address] !== undefined) {
+      if (tokenMap[tokenInfo.chainId][getFormattedAddress(tokenInfo.address)] !== undefined) {
```
Comment (hungdoansy) [inline src/state/lists/hooks.ts:59]: Too long expression here. Please split them into parts

File: src/state/lists/hooks.ts [hunk 3]
```diff
@@ -55,11 +56,11 @@ function listToTokenMap(list: TokenList): TokenAddressMap {
...
         return tokenMap
       }
       const token = new WrappedTokenInfo(tokenInfo, list)
-      tokenMap[tokenInfo.chainId][tokenInfo.address] = token
+      tokenMap[tokenInfo.chainId][getFormattedAddress(tokenInfo.address)] = token
```
Comment (hungdoansy) [inline src/state/lists/hooks.ts:63]: Too long expression here. Please split them into parts

File: src/state/lists/hooks.ts (hunk 2 above)
Comment (namgold) [reply src/state/lists/hooks.ts:59]: how would you suggest?

Comment (hungdoansy) [reply src/state/lists/hooks.ts:59]:
```ts
  const map = list.tokens.reduce<TokenAddressMapWriteable>((tokenMap, tokenInfo) => {
    const tokenAddress = getFormattedAddress(tokenInfo.address)

    if (tokenMap[tokenInfo.chainId][tokenAddress] !== undefined) {
      return tokenMap
    }

    const token = new WrappedTokenInfo(tokenInfo, list)
    tokenMap[tokenInfo.chainId][tokenAddress] = token
    return tokenMap
  }, EMPTY_LIST())
```

or even better, you can take chainId out of tokenInfo

Comment (hungdoansy) [reply src/state/lists/hooks.ts:59]: I removed spread in EMPTY_LIST() as it's not needed

Comment (namgold) [reply src/state/lists/hooks.ts]: done

**Review by hungdoansy — APPROVED** [review]
LGTM

**Pull request merged**